source/include/backoff_algorithm.h

```diff
@@ -135,15 +111,22 @@ void BackoffAlgorithm_InitializeParams( BackoffAlgorithmContext_t * pContext,
  *
  * @param[in, out] pRetryContext Structure containing parameters for the next backoff
  * value calculation.
+ * @param[in] randomValue The random value to use for calculation of the backoff period.
+ * The random value should be in the range of [0, UINT32_MAX].
  * @param[out] pNextBackOff This will be populated with the backoff value (in milliseconds)
  * for the next retry attempt. The value does not exceed the maximum backoff delay
  * configured in the context.
  *
- * @return #BackoffAlgorithmSuccess after a successful sleep, #BackoffAlgorithmRngFailure for a failure
- * in random number generation, #BackoffAlgorithmRetriesExhausted when all attempts are exhausted.
+ * @note It is recommended that a True Random Number Generator (TRNG) is used to generate
+ * the random value. Using a random number generator that generates device-specific unique values
+ * specific mitigates possibility of collision between multiple devices retrying the network operations.
```

Comment:
This last sentence seems problematic to me.  A TRNG does not generate unique values, it generates random values.  The entropy pool may be device specific, but conceptually the random number is not device-specific; it's just random.

Reply:
Yes I agree we should just say that using random adds diversity, too much detail here is just a trap, random is hard.